### ISSUE TITLE
Use Bliss.bind throughout instead of addEventListener

### ISSUE THREAD

**Issue by hopeful2** [mavoweb/mavo#415]
Mavo used to include Bliss Full which can remove listeners added by Bliss.bind and elem.addEventListener.

Now Mavo has changed to Bliss Shy and will remove listeners added by Bliss.bind method only. 

In an effort to release memory when destroying a Mavo instance, we need to remove all listeners and references to the instance, nodes/elements belonging to the instance. 

I am proposing to use Bliss.bind($.bind) throughout the code instead of addEventListener. This is to track all listeners by elements and remove them by batch when necessary. The few exceptions are  listeners added to window or document object which will always stay there or will be removed automatically. 

In my local copy I have replaced addEventListener with Bliss.bind and Mavo seems to work fine. Bliss.bind does some extra housekeeping and calls native addEventListener anyway. 

If these changes are desirable in the core, I will clean up the local copy and submit a PR.

**Comment by LeaVerou** [comment]
Hi there,

Thanks again for your work!
I'm skeptical. In general I like using the native methods when the Bliss ones are not needed. However, if that causes memory leaks, it clearly needs fixing. However, are you sure it does, or are you hoping that's the source of the memory issues? It seems odd that these listeners would be never garbage collected even after the elements are not in the DOM anymore and there are no references left to them.

**Comment by hopeful2** [comment]
I agree with you in principle to use native methods and achieve the same outcome. However there is no way to remove listeners with anonymous functions as event handler using the native addEventListener method.  Bliss does the job better to manage listeners with anonymous functions in this case.

The context of most anonymous functions contains references to Mavo instance/children nodes/collection items ("this"), and the Mavo instance/children nodes/collection items contain references to attached and detached elements. The elements contain reference to anonymous functions through registered event listeners. Because of these circular references, when the elements are removed from the DOM, they become detached elements/nodes, but still remain in the memory. They can not be garbage collected. This is my finding after days of monitoring the memory allocation and retainers in Chrome dev tools.  Removing event listeners with anonymous functions will break the circular reference and release the allocated resources. 

The challenge on memory management is that a single surviving reference can ruin the whole effort.  It is either all or none. 

Your thoughts?
